### ISSUE TITLE
Document or remove path_sep_replace config value

### ISSUE THREAD

**Issue by mrjoel** [beetbox/beets#323]
I struggled for quite a while to not have slashes replaced with underscores (I use a comma for readability). Based on the documentation of the configuration (which is otherwise quite thorough, thanks!), I tried replacing the regex in the replace configuration value. Only after digging at the code did I find that there is a separate path_sep_replace config value used to define what to use for replacement.

I didn't look in detail, but it looks like as long as a valid regex replacement value is given then the separate config value isn't needed. If it is needed, a quick mention during the discussion of the replace configuration section would be quite helpful.

**Comment by pnelson** [comment]
I ran into this one today. It has been a while since the original report or references. Have you given any more thought on this?

**Comment by sampsyo** [comment]
I do think about it from time to time, but I'm still convinced there's no good solution: the current behavior is confusing, and the ideal behavior is surprisingly hard to get right. Bright ideas are always welcome.

**Comment by petwri** [comment]
So, this feature is apparently still implemented, [it](https://github.com/beetbox/beets/blob/master/beets/dbcore/db.py) has the relevant parts still there:

`        if self.for_path:
            sep_repl = beets.config['path_sep_replace'].get(unicode)
            for sep in (os.path.sep, os.path.altsep):
                if sep:
                    value = value.replace(sep, sep_repl)`

It should replace any occurence of "/" on my linux system with the 'path_sep_replace' from my config. But, for some reason, it is not. Any ideas?

**Comment by sampsyo** [comment]
@petwri We might be able to help, but we'll need more details (i.e., the standard bug report stuff including a specific way to reproduce the problem).

**Comment by petwri** [comment]
@sampsyo Sorry I am coming back to this after not paying any attention to it for quite a while, I just found out that it is still in the code (see my posting from 8 May). But I can't really tell why "sep" doesn't get replaced by "sep_rep1". If you need, I could provide more info.

**Comment by sampsyo** [comment]
Yes, please provide a full bug report. A new ticket is probably in order.

**Comment by petwri** [comment]
Will asap.

**Comment by petwri** [comment]
So, tried it again, importing and moving resulted in this:

`main@htpc:~/.scripts$ beet ls -f '$path'
** plugin echonest not found
/media/Music/Steven Wilson/4 1/2/01-01.mp3
/media/Music/Steven Wilson/4 1/2/01-02.mp3
/media/Music/Steven Wilson/4 1/2/01-03.mp3
/media/Music/Steven Wilson/4 1/2/01-04.mp3
/media/Music/Steven Wilson/4 1/2/01-05.mp3
/media/Music/Steven Wilson/4 1/2/01-06.mp3`

Version is 1.3.18, Python 2.7.11+, Linux 4.5.2-040502-generic

[beets-logfile.txt](https://github.com/beetbox/beets/files/360201/beets-logfile.txt)

[config.yaml.txt](https://github.com/beetbox/beets/files/360205/config.yaml.txt)

**Comment by sampsyo** [comment]
Huh, that's odd. This is a separate issue, though—can you please open a new bug so we don't bother the subscribers to this issue?

Then, it might be useful to see if this happens in the default configuration, with everything turned off. Out of the box, beets should replace slashes automatically.

**Comment by petwri** [comment]
When you're talking about default config, where do I get that from?

**Comment by jackwilsdon** [comment]
You can just (temporarily) move your `config.yaml` elsewhere which should set the configuration back to default. 😄

**Comment by petwri** [comment]
Issue created: https://github.com/beetbox/beets/issues/2126
